test/pdmtypes.jl

```diff
@@ -296,4 +304,15 @@ using Test
         @test_broken C - B isa Diagonal{Float64, Vector{Float64}}
         @test C - B ≈ Matrix(C) - Matrix(B)
     end
+
+    # Ref https://github.com/JuliaStats/PDMats.jl/pull/207
+    # Note: `cholesky(::SymTridiagonal)` requires https://github.com/JuliaLang/julia/pull/44076
+    if VERSION >= v"1.8.0-DEV.1526"
+        @testset "PDMat from SymTridiagonal" begin
+            S = SymTridiagonal(fill(4, 4), fill(1, 3))
+            M = @inferred(PDMat(S))
+            @test M isa PDMat{Int,<:SymTridiagonal,<:Cholesky}
+            @test M == S
+        end
+    end
```

Comment:
Probably want to add tests for what happens if one tries to construct with factorizations besides cholesky.